feature-libs/product-multi-dimensional/selector/core/services/product-multi-dimensional-selector-images.service.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { TestBed } from '@angular/core/testing';
+import { Config, VariantOptionQualifier } from '@spartacus/core';
+import { ProductMultiDimensionalSelectorImagesService } from './product-multi-dimensional-selector-images.service';
+import { ProductMultiDimensionalConfig } from '../../root/config/product-multi-dimensional-config';
+
+enum VariantQualifier {
+  STYLE_SWATCH = 'styleSwatch',
+}
+
+describe('ProductMultiDimensionalSelectorImagesService', () => {
+  let service: ProductMultiDimensionalSelectorImagesService;
+  let config: Config;
+
+  beforeEach(() => {
+    const mockConfig = {
+      backend: {
+        media: {
+          baseUrl: 'http://media.base.url/',
+        },
+        occ: {
+          baseUrl: 'http://occ.base.url/',
+        },
+      },
+      multiDimensional: {
+        imageFormat: VariantQualifier.STYLE_SWATCH,
+      } as ProductMultiDimensionalConfig,
+    };
+
+    TestBed.configureTestingModule({
+      providers: [
+        ProductMultiDimensionalSelectorImagesService,
+        { provide: Config, useValue: mockConfig },
+      ],
+    });
+
+    service = TestBed.inject(ProductMultiDimensionalSelectorImagesService);
+    config = TestBed.inject(Config);
+  });
+
+  describe('getVariantOptionImages', () => {
+    it('should return images that match the specified format', () => {
+      config.multiDimensional.imageFormat = VariantQualifier.STYLE_SWATCH;
+      const variantOptionQualifiers: VariantOptionQualifier[] = [
+        {
+          image: { format: VariantQualifier.STYLE_SWATCH, url: '/image1.jpg' },
+        },
+        {
+          image: { format: 'large', url: '/image2.jpg' },
+        },
+      ];
+      const variantOptionValue = 'Sample Variant';
+
+      const result = service.getVariantOptionImages(
+        variantOptionQualifiers,
+        variantOptionValue
+      );
+
+      expect(result.length).toBe(1);
+      expect(result[0]).toEqual({
+        altText: variantOptionValue,
+        format: VariantQualifier.STYLE_SWATCH,
+        url: 'http://media.base.url//image1.jpg',
+      });
+    });
+
+    it('should use altText from image if available', () => {
+      config.multiDimensional.imageFormat = VariantQualifier.STYLE_SWATCH;
```

Comment:
```suggestion
      config.multiDimensional = { imageFormat: VariantQualifier.STYLE_SWATCH };
```
To avoid error `'config.multiDimensional' is possibly 'undefined'.`

Reply:
fixed